cmd/server.go

```diff
@@ -20,44 +20,134 @@ import (
 	"context"
 	"crypto/tls"
 	"fmt"
+	dockhand "github.com/boxboat/dockhand-secrets-operator/pkg/apis/dockhand.boxboat.io/v1alpha1"
 	"github.com/boxboat/dockhand-secrets-operator/pkg/common"
 	"github.com/boxboat/dockhand-secrets-operator/pkg/k8s"
 	"github.com/boxboat/dockhand-secrets-operator/pkg/webhook"
 	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/leaderelection"
 	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 )
 
 type ServerArgs struct {
-	serverPort        int
-	serverCert        string
-	serverKey         string
-	serviceName       string
-	serviceNamespace  string
-	selfSignCerts     bool
+	serverPort       int
+	serverCert       string
+	serverKey        string
+	serviceName      string
+	serviceId        string
+	serviceNamespace string
+	selfSignCerts    bool
 }
 
 var (
 	serverArgs ServerArgs
 )
 
-func runServer(ctx context.Context) {
+func certManager() {
+	lock, err := k8s.GetLeaseLock(serverArgs.serviceId, serverArgs.serviceName, serverArgs.serviceNamespace)
+	common.ExitIfError(err)
+
+	leaderCtx, cancel := context.WithCancel(context.Background())
```

Comment:
You could pass in a context to the `certManager` function and use that here

Reply:
Updated